src/App.jsx

```diff
@@ -1,11 +1,127 @@
+            console.error('Error fetching country:', error);
+            setError('Er is een fout opgetreden. Probeer het opnieuw.');
+        }
+    };
+
+
+    const handleSearch = async () => {
+        if (searchTerm) {
+            await fetchCountry(searchTerm);
+            setSearchTerm(''); //Leeg het zoekveld na zoekopdracht
+        }
+    };
+
+    const handleKeyDown = (event) => {
+        if (event.key === 'Enter') {
+            handleSearch();
+        }
+    };
+
     return (
-        <>
-            Maak je applicatie hier!
-        </>
+        <div className="App">
+            <h1>Country Information</h1>
+            <img src={worldMap} alt="WorldMap" style={{ width: '100%', maxHeight: '500px', objectFit: 'cover' }} />
+
+            {/*} //Functionaliteit Opdracht 1 - weergeven van alle landen
+            {!isFetched && (
+                <button onClick={fetchCountries}>Fetch Countries</button>)}
+
+            <div className="countries-container">
+                {countries.map((country) => (
+                    Voor het identificeren van landen in deze .map functie wordt de unieke cca3 code die elk land heeft gebruikt. Dit zorgt ervoor dat React efficienter met de lijst omgaat.
+                    <div key={country.cca3}>
+                        <img src={country.flags.svg} alt={`Flag of ${country.name.common}`} width="100" />
+                        <p style={{ color: getColorByRegion(country.region) }}>{country.name.common}</p>
+                        <p>Has a population of {country.population.toLocaleString()} people</p>
+                    </div>
+                ))}
+            </div>
+        */}
+
+            {/* Functionaliteit Opdracht 2 */
+            }
+            <div className="search-container">
+                <input
+                    type="text"
+                    value={searchTerm}
+                    onChange={(e) => setSearchTerm(e.target.value)}
+                    onKeyDown={handleKeyDown}
+                    placeholder="Search for a country..."
+                    />
+                <button onClick={handleSearch}>Search</button>
+            </div>
+
+            {error && <p className="error">{error}</p>}
+
+            {countryData && (
+                <div className="country-info">
+                    <img src={countryData.flags.svg} alt={`Flag of ${countryData.name.common}`} width="100" />
+                    <h2 style={{ color: getColorByRegion(countryData.region) }}>{countryData.name.common}</h2>
+                    <p>{countryData.name.common} is situated in {countryData.subregion} and the capital is {countryData.capital[0]}</p>
+                    <p>It has a population of {convertPopulationToMillions(countryData.population)} million people and it borders with {countryData.borders ? countryData.borders.length : 0} neighboring countries</p>
+                    <p>Websites can be found on {countryData.domains.join(',')} domain&amp;s</p>
```

Comment:
Kijk nog eens goed met een console.log naar het object die je binnenhaalt vanuit de api. Je zult vinden dat er binnen het object geen key bestaat met de naam domains. Hierdoor gaat countryData.domains natuurlijk nooit werken. tld is de naam van de key die het domein bevat. {countryData.tld} zou correct zijn.